prompting/rewards/date.py

```diff
@@ -12,37 +15,50 @@ def name(self) -> str:
     def __init__(self, **kwargs):
         super().__init__()
 
+
+    
+    def date_diff(self, ref_date, comp_date):
+        """
+        Calculates the absolute difference in days between two dates.
+        """
+        return abs(ref_date[0] - comp_date[0]).days + 365*abs(int(ref_date[1]) - int(comp_date[1]))
+    
+    def parse_dates_from_text(self, text: str):
+        """
+        Parses dates from a body of text, handling various formats, and returns pandas datetime objects.
+        """
+
+        date_patterns = [
+            r"\b(\d{1,2})[/-](\d{1,2})[/-](\d{3,4})\b",  # MM/DD/YYYY or DD/MM/YYYY
```

Comment:
```suggestion
            r"\b(\d{1,2})[/-](\d{1,2})[/-](\d{2,4})\b",  # MM/DD/YYYY or DD/MM/YYYY
```
 makes more sense to expect YY or YYYY rather than YYY

Reply:
This is for the 834 AD years and stuff.

Reply:
And looking for YY means that some DD or MM may mmess up hte pattern.